### PR TITLE
replace grunt-autoprefixer with gruntpostcss + autoprefixer

### PR DESCRIPTION
**Pull request by bassjobsen**

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -21,10 +21,22 @@ module.exports = function (grunt) {
   var isTravis = require('is-travis');
   var npmShrinkwrap = require('npm-shrinkwrap');
   var mq4HoverShim = require('mq4-hover-shim');
+  var autoprefixer = require('autoprefixer');
 
   var generateCommonJSModule = require('./grunt/bs-commonjs-generator.js');
   var configBridge = grunt.file.readJSON('./grunt/configBridge.json', { encoding: 'utf8' });
 
+  var browsers = [
+          'Android 2.3',
```
Comment (cvrebert) [inline Gruntfile.js]: Use 2 spaces for indents

File: Gruntfile.js [hunk 2]
```diff
@@ -46,8 +58,8 @@ module.exports = function (grunt) {
                  '}\n',
     jqueryVersionCheck: '+function ($) {\n' +
                         '  var version = $.fn.jquery.split(\' \')[0].split(\'.\')\n' +
-                        '  if (version[0] !== \'2\') {\n' +
-                        '    throw new Error(\'Bootstrap\\\'s JavaScript requires jQuery version 2.x.x\')\n' +
+                        '  if ((version[0] < 2 && version[1] < 9) || (version[0] == 1 && version[1] == 9 && version[2] < 1)) {\n' +
```
Comment (cvrebert) [inline Gruntfile.js]: Undo this change

**Comment by bassjobsen** [comment]
yes thanks, i see. Done.

**Comment by XhmikosR** [comment]
Any change in dist files?

**Comment by bassjobsen** [comment]
dist files are unchanged (as expected)

**Comment by XhmikosR** [comment]
Then it should be OK.

File: Gruntfile.js [hunk 3]
```diff
@@ -215,38 +227,30 @@ module.exports = function (grunt) {
...
-          'Opera >= 12',
-          'Safari >= 7.1'
-        ]
-      },
       core: {
         options: {
-          map: true
+          map: true,
+          processors: [
+          mq4HoverShim.postprocessorFor({ hoverSelectorPrefix: '.bs-true-hover ' }),
```
Comment (cvrebert) [inline Gruntfile.js]: needs to be indented another level

File: Gruntfile.js [hunk 4]
```diff
@@ -215,38 +227,30 @@ module.exports = function (grunt) {
...
+          autoprefixer({ browsers: browsers })
+          ]
         },
         src: 'dist/css/*.css'
       },
       docs: {
+        options: {
+          processors: [
+          autoprefixer({ browsers: browsers })
```
Comment (cvrebert) [inline Gruntfile.js]: needs to be indented another level

File: Gruntfile.js [hunk 5]
```diff
@@ -215,38 +227,30 @@ module.exports = function (grunt) {
...
+          ]
+        },
         src: 'docs/assets/css/docs.min.css'
       },
       examples: {
+        options: {
+          processors: [
+          autoprefixer({ browsers: browsers })
```
Comment (cvrebert) [inline Gruntfile.js]: needs to be indented another level

**Comment by cvrebert** [comment]
Modulo the indentation, LGTM.
We'll need to reshrinkwrap after merging this.

**Comment by XhmikosR** [comment]
@bassjobsen: can you fetch and rebase and also take care of @cvrebert's comments?

**Comment by bassjobsen** [comment]
i will try it this weekend. I'm not sure if i do understand the rebase thing.

**Comment by bassjobsen** [comment]
Well i tried / try to do the rebase.
i'm not sure i did it right.
I also found some diffs in `dist` now:

        4893,4894c4893
        <        -o-transition:         transform .3s ease-out, -o-transform .3s ease-out;
        <           transition: -webkit-transform .3s ease-out;
        ---
        >        -o-transition:      -o-transform .3s ease-out;
        4896d4894
        <           transition:         transform .3s ease-out, -webkit-transform .3s ease-out, -o-transform .3s ease-out;
        5289,5290c5287
        <          -o-transition:         transform .6s ease-in-out, -o-transform .6s ease-in-out;
        <             transition: -webkit-transform .6s ease-in-out;
        ---
        >          -o-transition:      -o-transform .6s ease-in-out;
        5292d5288
        <             transition:         transform .6s ease-in-out, -webkit-transform .6s ease-in-out, -o-transform .6s ease-in-out;

**Comment by bassjobsen** [comment]
i do not understand why 31 files are change now? @XhmikosR should i better create a new PR now?

**Comment by wolfy1339** [comment]
You didn't do the rebase properly, which would account for the extra unwanted commits and file changes.
I would make a new one, it's easier and faster.

**Comment by kkirsche** [comment]
I'd agree that it's easier and faster to make a new one @bassjobsen. I'd recommend this guide on how to rebase https://help.github.com/articles/about-git-rebase/ if you aren't familiar with it / don't do it often

**Comment by XhmikosR** [comment]
`git rebase -i upstream/v4-dev`

**Comment by kkirsche** [comment]
@XhmikosR for the sake of being complete, `git fetch upstream/v4-dev && git rebase -i upstream/v4-dev`

**Comment by bassjobsen** [comment]
okay thanks, i will create a new PR soon

**Comment by XhmikosR** [comment]
That is not in order to create a new PR...

**Comment by bassjobsen** [comment]
@XhmikosR yes i understand, i will repeat my changes on the latest version and add it to a new PR

**Comment by bassjobsen** [comment]
@XhmikosR maybe it is rebased now?

**Comment by XhmikosR** [comment]
@bassjobsen: yeah it's good now. But you should squash some of the patches while rebasing.

**Comment by wolfy1339** [comment]
commit 17f7b5c4788aedaa161d9cbfd79cc12f02b835e1 includes unwanted changes (from improper rebase)

**Comment by XhmikosR** [comment]
Good catch, indeed it does.

**Comment by bassjobsen** [comment]
@wolfy1339 yes i see, but https://github.com/bassjobsen/bootstrap/commit/52b5bd5ce99e2c125b16939ea6e0101381756b55 repairs this?

**Comment by wolfy1339** [comment]
That commit won't cut it either. You would need to rewrite the git history of your branch to only have your changes.
When you do a rebase, you need to do a force push by using ```git push -f```, so the git history is replaced and nothing unwanted happens.
Since you aren't comfortable with rebase, I would recommend that you create a new pull request and not that it superseded this one.

**Comment by cvrebert** [comment]
Leave this open. I'm rebasing and merging this myself.

**Comment by cvrebert** [comment]
@XhmikosR This is why I always recommend qualifying requests for rebases with "if you are comfortable+familiar with rebasing..."

**Comment by cvrebert** [comment]
Rebased. Running Travis build: https://travis-ci.org/twbs/bootstrap/builds/90641415

**Comment by cvrebert** [comment]
Build failed. Retrying now that #17652 has been merged.
https://travis-ci.org/twbs/bootstrap/builds/90642551

**Comment by cvrebert** [comment]
There seems to be a problem installing `phantomjs@1.9.18` under Node.js v4.2.2 & `npm@2.14.7`.

**Comment by cvrebert** [comment]
Forcing npm@3 fixes that problem, but now some unrelated random RVM problem has raised its head.

**Comment by cvrebert** [comment]
Filed https://github.com/travis-ci/travis-ci/issues/5092. Travis is missing curl for some reason, which causes the RVM problem.

**Comment by XhmikosR** [comment]
Why did you need to upgrade Ruby anyway? I use 2.2 in another project of mine without any issues.

**Comment by XhmikosR** [comment]
```yml
env:
  matrix:
    - RUBY_VERSION=2.2

before_install:
  - rvm install $RUBY_VERSION
  - rvm use $RUBY_VERSION --fuzzy
  - export GEMDIR=$(rvm gemdir)
```
@cvrebert: try this, it should work.

**Comment by XhmikosR** [comment]
Also, we shouldn't force npm@3 either. I mean, I'm using 2.x locally just fine. Did you try to clear Travis cache just in case?

**Comment by XhmikosR** [comment]
@cvrebert: this seems to pass https://github.com/twbs/bootstrap/commit/9b50133387b688a058844693ce13c9a0d0ed08ad
https://travis-ci.org/twbs/bootstrap/builds/90675687

**Comment by cvrebert** [comment]
Didn't really need to upgrade Ruby. I was just grasping at straws when debugging the RVM issue.

> this seems to pass 9b50133
> https://travis-ci.org/twbs/bootstrap/builds/90675687

@XhmikosR Looks good, you should open a PR for that.

**Comment by cvrebert** [comment]
No, haven't tried clearing the Travis cache.

**Comment by XhmikosR** [comment]
Shouldn't I just merge that then?

**Comment by cvrebert** [comment]
Nicely done: 7b19dfc06fc9c4f5d964c0e9a596843e9a4b4f33

**Comment by cvrebert** [comment]
For whatever reason, the RVM error is no longer occurring.

**Comment by cvrebert** [comment]
Purged the caches and building one more time to verify whether npm@3 is truly necessary:
https://travis-ci.org/twbs/bootstrap/builds/90687640

**Comment by cvrebert** [comment]
Build failed. @XhmikosR Yup, npm@3 is needed to install from an npm@3 shrinkwrap, which is reasonable. I'm guessing you don't see it locally since you're probably not installing via shrinkwrap.

**Comment by XhmikosR** [comment]
@cvrebert: we don't use a v3.x shrinkwrap, do we?

**Comment by cvrebert** [comment]
@XhmikosR npm v3 is the latest stable major version of npm and seems to the preinstalled version for Node.js v5. And it's compatible with Node.js v4. Since the shrinkwrap is only used internally, I don't see much reason to use an old version.
(My branch switches to npm v3 shrinkwrap since I have Node.js v5 on my local box.)

**Comment by cvrebert** [comment]
Rebased and merged as 50c43bc55a3576636cf83eae2a918fa6dd231b75. Thanks @bassjobsen!

@XhmikosR Feel free to argue about 3236a4df6971836111b768302b7d464fb457a0e0 tomorrow :smile:

**Comment by XhmikosR** [comment]
The thing is, I don't want to use v5.0.0 yet... This broke my branch and I will need to use npm 3.x to rebase it.
